src/utils/getUrlByRequestOptions.ts

```diff
@@ -16,11 +17,18 @@ export function getUrlByRequestOptions(
 
   debug('creating URL from options:', options)
 
+  // Inherit the protocol from the Agent, if present.
+  if (options.agent instanceof Agent) {
```

Comment:
`agent: Agent | boolean`. To prevent false positive with `agent && agent.protocol` (where `agent` may be `true` and `true.protocol` throws`) I'm explicitly checking if it's an instance of `http.Agent`. `https.Agent` extends `http.Agent`, so the check would pass for it as well.